src/averageAges.js

```diff
@@ -55,7 +68,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.map(person => (
+    [person.name, person.mother, person.sex, person.born]
+  ));
+  const mothers = children.map((child, i, arr) => (
+    arr.find(person => person[0] === child[1])
+  ));
+  const totalInfo = children.map((child, i) => (
+    [...child, mothers[i] !== undefined ? mothers[i][3] : undefined]
+  ));
+  const totalInfoFiltered = totalInfo.filter(years => (
+    years[4] !== undefined && (!onlyWithSon || years[2] === 'm')
+  ));
+  const sumOfDiff = totalInfoFiltered.reduce((accumulator, years) => (
+    accumulator + (years[3] - years[4])
+  ), 0);
+
+  return sumOfDiff / totalInfoFiltered.length;
```

Comment:
I think you can shorten this part, using 'find' method:)

Reply:
Thank you for your advice. Now I've implemented it